test/utils.py

```diff
@@ -1,17 +1,251 @@
+        this.push(config)
+        return this
+
+    def authz_client(self) -> SyncClient:
+        return self.current.authz_client()
+
+    def authz_async_client(self) -> AsyncClient:
+        return self.current.authz_async_client()
+
+    def push(self, config: AuthzConfig):
+        self.stack.append(config)
+
+    def pop(self) -> AuthzConfig:
+        return self.stack.pop()
+
+
+@dataclass
+class NonCachingAuthz(Authz):
+    @property
+    def client(self) -> AsyncClient:
+        return self.authz_config.authz_async_client()
+
+
+@dataclass
+class TestAppConfig(Config):
+    """Test class that can handle isolated dbs and authz instances."""
+
+    @classmethod
+    def from_env(cls, prefix: str = "") -> "Config":
+        """Create a config from environment variables."""
+
+        user_store: base_models.UserStore
+        authenticator: base_models.Authenticator
+        gitlab_authenticator: base_models.Authenticator
+        gitlab_client: base_models.GitlabAPIProtocol
+        user_preferences_config: UserPreferencesConfig
+        version = os.environ.get(f"{prefix}VERSION", "0.0.1")
+        server_options_file = os.environ.get("SERVER_OPTIONS")
+        server_defaults_file = os.environ.get("SERVER_DEFAULTS")
+        k8s_namespace = os.environ.get("K8S_NAMESPACE", "default")
+        max_pinned_projects = int(os.environ.get(f"{prefix}MAX_PINNED_PROJECTS", "10"))
+        user_preferences_config = UserPreferencesConfig(max_pinned_projects=max_pinned_projects)
+        db = DBConfigStack.from_env(prefix)
+        kc_api: IKeycloakAPI
+        secrets_service_public_key: PublicKeyTypes
+        gitlab_url: str | None
+
+        encryption_key = secrets.token_bytes(32)
+        secrets_service_public_key_path = os.getenv(f"{prefix}SECRETS_SERVICE_PUBLIC_KEY_PATH")
+        if secrets_service_public_key_path is not None:
+            secrets_service_public_key = serialization.load_pem_public_key(
+                Path(secrets_service_public_key_path).read_bytes()
+            )
+        else:
+            private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+            secrets_service_public_key = private_key.public_key()
+
+        authenticator = DummyAuthenticator()
+        gitlab_authenticator = DummyAuthenticator()
+        quota_repo = QuotaRepository(DummyCoreClient({}, {}), DummySchedulingClient({}), namespace=k8s_namespace)
+        user_always_exists = os.environ.get("DUMMY_USERSTORE_USER_ALWAYS_EXISTS", "true").lower() == "true"
+        user_store = DummyUserStore(user_always_exists=user_always_exists)
+        gitlab_client = DummyGitlabAPI()
+        dummy_users = [
+            user_preferences_models.UnsavedUserInfo(
+                id="user1", first_name="user1", last_name="doe", email="user1@doe.com"
+            ),
+            user_preferences_models.UnsavedUserInfo(
+                id="user2", first_name="user2", last_name="doe", email="user2@doe.com"
+            ),
+        ]
```

Comment:
suggestion(non-blocking): I think these were hardcoded before but it would be nicer if these were also fixtures and they were added as part of the fixture that initializes the TestAppConfig.